### DISCUSSION TITLE
Can I squelch error messages?

### DISCUSSION THREAD

**Discussion by aaronkyle** [observablehq/framework#1811] [Q&A] [answered]
I am trying to avoid showing red error messages by creating conditional views that only show when a file is loaded.  However despite my attempts, I am still seeing them:

![Screenshot from 2024-11-11 09-35-31](https://github.com/user-attachments/assets/783a2fdb-8402-4cdf-af4e-abc7c426784d)

Is there a way to turn these off?

**Comment by mythmon** [comment]
I don't know of any way to disable the error display, but you could handle the error instead. For example, you could use a try catch: 

~~~js
```js
const myValue = (() => {
  try {
    return file.csv();
  } catch (error) {
    console.warn("error with file", error);
    return null;
  }
})();
```
~~~

or maybe just use a nullish coalescing operator:

~~~js
```js
const myValue = file?.csv();
```
~~~

**Comment by aaronkyle** [comment]
Thank you, @mythmon .  The error handling could work to return a blank/template data object, which would satisfy my downstream functions enough that they also wouldn't error out.  If there were a simple means of switching error reporting from the display to instead only the console, it would be a lot easier.

**Comment by mbostock** [comment, marked as answer]
If you set the **required** option on the Inputs.file, it will make the default value `undefined` rather than `null`, which will cause downstream cells to wait rather than exposing `file` as null.

**Comment by aaronkyle** [comment]
Perfect! Thank you, @mbostock !
